playpen/mongoengine/validation_check.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python2
+
+import time
+import json
+import os
+import sys
+from subprocess import call
+
+from bson.json_util import dumps
+from mongoengine import ValidationError
+
+from pulp.server.db import model as platform_models
+from pulp_rpm.plugins.db import models as rpm_models
+from pulp_docker.plugins import models as docker_models
+from pulp_puppet.plugins.db import models as puppet_models
+from pulp_ostree.plugins.db import model as ostree_models
+from pulp.server import config
+from pulp.server.db import connection as db_connection
+
+VALIDATION_ERROR_MESSAGE = "There were validation errors.  Please see {0} for more information\n"
+ERROR_FILE_NAME = "validation_errors_{0}.txt"
+MONGOENGINE_MODELS = [platform_models.Repository,
+                      platform_models.RepositoryContentUnit,
+                      platform_models.Importer,
+                      platform_models.ReservedResource,
+                      platform_models.Worker,
+                      platform_models.MigrationTracker,
+                      platform_models.TaskStatus,
+                      platform_models.CeleryBeatLock,
+                      platform_models.User,
+                      platform_models.Distributor,
+                      rpm_models.Distribution,
+                      rpm_models.DRPM,
+                      rpm_models.RPM,
+                      rpm_models.SRPM,
+                      rpm_models.Errata,
+                      rpm_models.PackageGroup,
+                      rpm_models.PackageCategory,
+                      rpm_models.PackageEnvironment,
+                      rpm_models.YumMetadataFile,
+                      rpm_models.ISO,
+                      docker_models.Blob,
+                      docker_models.Image,
+                      docker_models.Manifest,
+                      puppet_models.Module,
+                      ostree_models.Branch]
+
+
+class ValidationExceptionHandler:
+    """
+    Handles the output of validation exceptions to the exceptions file.
+    """
+    def __init__(self, error_file):
+        """
+        Initializes a ValidationException handler that writes to the specified error_file
+
+        :param error_file: The file to write exceptions to
+        :type error_file: file
+        """
+        self.error_file = error_file
+
+    def _write_newline(self):
+        self.error_file.write(os.linesep)
+
+    def handle_exception(self, item, ex):
+        """
+        Writes the validation error that was thrown to the error file specified in the constructor
+
+        :param item: The object that caused the validation exeption
+        :type item: mongoengine.Document
+        :param ex: Validation Error thrown by mongoengine
```

Comment:
I usually put a newline in between params so `param` and `type` statements are grouped per variable.